src/Tools/CoreFx.Tools/GenerateSlnFile.cs

```diff
@@ -0,0 +1,497 @@
+                string fileTemplateContent = File.ReadAllText(TemplateFile);
+                //not including ref
+                //TODO:Add ref back
+                //string pathToRef = "ref";
+                string pathToSrc = "src";
+                string pathToTest = "tests";
+                //will not work on Linux (expected)
+                string pathToPerfTest = "tests\\Performance";
+
+                //order of projects
+                List<string> projectTypeOrder = new List<string>() { pathToTest, pathToPerfTest, pathToSrc };
+
+                Dictionary<string, ProjectType> mapOfProjectTypes = new Dictionary<string, ProjectType>();
+
+                mapOfProjectTypes.Add(pathToSrc, new ProjectType());
+                mapOfProjectTypes.Add(pathToTest, new ProjectType(new List<string>() { "src" }, true, new List<string> { "Performance" }));
+                mapOfProjectTypes.Add(pathToPerfTest, new ProjectType(new List<string>() { "src" }));
+
+                Dictionary<string, List<string>> headers = new Dictionary<string, List<string>>();
+                string header = string.Empty;
+
+                Dictionary<string, List<SlnProject>> mapOfProjectsToTypes = new Dictionary<string, List<SlnProject>>();
+
+                Dictionary<string, ProjectType> transientMapOfProjectTypes = new Dictionary<string, ProjectType>();
+                foreach (var projectType in mapOfProjectTypes)
+                {
+                    string currentDirectory = Path.Combine(slnFile, projectType.Key);
+                    ProjectType currentProjectType = projectType.Value;
+                    currentProjectType.Type = projectType.Key;
+                    if (Directory.Exists(currentDirectory))
+                    {
+                        List<string> projFiles =
+                            Directory.EnumerateFiles(currentDirectory, "*proj", currentProjectType.SearchRecursively ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly).ToList();
+                        foreach (var projFile in projFiles)
+                        {
+                            if (currentProjectType.ExclusionStrings != null)
+                            {
+                                if (currentProjectType.ExclusionStrings.Any(x => projFile.Contains(x)))
+                                {
+                                    continue;
+                                }
+                            }
+                            currentProjectType.InUse = true;
+                            SlnProject currentProject = new SlnProject(ReadProjectGuid(projFile));
+                            string configurationProps = Path.Combine(Path.GetDirectoryName(projFile), CONFIGURATION_PROPS);
+                            currentProject.configurations = ReadConfigurations(configurationProps);
+                            currentProject.Type = projectType.Key;
+                            List<SlnProject> projectGuids = new List<SlnProject>();
```

Comment:
If there's already a list in the `Dictionary`, then you're just throwing this new list out. You can put this constructor call down below in the `else` block.